documentation/modules/mirrormaker2/proc-mirrormaker-replication.adoc

```diff
@@ -0,0 +1,125 @@
+// Module included in the following assemblies:
+//
+// assembly-mirrormaker.adoc
+
+[id='proc-mirrormaker-replication-{context}']
+= Synchronizing data between Kafka clusters using MirrorMaker 2.0
+
+Use MirrorMaker 2.0 to synchronize data between Kafka clusters through configuration.
+
+The previous version of MirrorMaker continues to be supported.
+If you wish to use the resources configured for the previous version,
+they must be updated to the format supported by MirrorMaker 2.0.
+
+The configuration must specify:
+
+* Each Kafka cluster
+* Connection information for each cluster, including TLS authentication
+* The replication flow and direction
+** Cluster to cluster
+** Topic to topic
+
+Use the properties of the `KafkaMirrorMaker2` resource to configure your Kafka MirrorMaker 2.0 deployment.
+
+MirrorMaker 2.0 provides default configuration values for properties such as replication factors.
+A minimal configuration, with defaults left unchanged, would be something like this example:
+
+[source,yaml,subs="+quotes,attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: KafkaMirrorMaker2
+metadata:
+  name: my-mirror-maker2
+  spec:
+    version: {DefaultKafkaVersion}
+    connectCluster: "my-cluster-target"
+    clusters:
+    - alias: "my-cluster-source"
+      bootstrapServers: my-cluster-source-kafka-bootstrap:9092
+    - alias: "my-cluster-target"
+      bootstrapServers: my-cluster-target-kafka-bootstrap:9092
+    mirrors:
+    - sourceCluster: "my-cluster-source"
+      targetCluster: "my-cluster-target"
+      sourceConnector: {}
+----
+
+You can configure access control for source and target clusters using TLS or SASL authentication.
+This procedure shows a configuration that uses TLS encryption and authentication for the source and target cluster.
```

Comment:
@ajborley We haven't actually shown this (yet). Do we have an example we can insert into the task?

Reply:
There's nothing in `examples/kafka-mirror-maker-2/` but here's an example from the system tests:

```
apiVersion: kafka.strimzi.io/v1alpha1
kind: KafkaMirrorMaker2
metadata:
  name: "my-mirror-maker2"
spec:
  replicas: 1
  version: 2.4.0
  connectCluster: "my-cluster-target"
  clusters:
  - alias: "my-cluster-source"
    authentication:
      certificateAndKey:
        certificate: source.crt
        key: source.key
        secretName: my-user-source
      type: tls
    bootstrapServers: my-cluster-source-kafka-bootstrap:9093
    tls:
      trustedCertificates:
      - certificate: ca.crt
        secretName: my-cluster-source-cluster-ca-cert
  - alias: "my-cluster-target"
    authentication:
      certificateAndKey:
        certificate: target.crt
        key: target.key
        secretName: my-user-target
      type: tls
    bootstrapServers: my-cluster-target-kafka-bootstrap:9093
    config:
      config.storage.replication.factor: 1
      offset.storage.replication.factor: 1
      status.storage.replication.factor: 1
    tls:
      trustedCertificates:
      - certificate: ca.crt
        secretName: my-cluster-target-cluster-ca-cert
  mirrors:
  - sourceCluster: my-cluster-source
    targetCluster: my-cluster-target
    sourceConnector:
      config:
        offset-syncs.topic.replication.factor: 1
        replication.factor: 1
        sync.topic.acls.enabled: "false"
    topicsPattern: .*
```

Reply:
@ajborley Thanks Andrew. I've added callouts similar to how we have them for the previous version of MirrorMaker, referring to the schema ref for more details.